src/entt/entity/handle.hpp

```diff
@@ -27,17 +27,39 @@ struct basic_handle {
         basic_registry<entity_type>
     >;
 
+    /**
+    * @brief Constructs a default invalid handle.
+    */
+    basic_handle() ENTT_NOEXCEPT
+        : basic_handle{null}
+    {}
+
+    /**
+    * @brief Constructs an invalid handle.
+    */
+    basic_handle(null_t) ENTT_NOEXCEPT
+        : basic_handle{nullptr, null}
+    {}
+
+    /**
+    * @brief Constructs a handle from another handle.
+    * @param other Another handle
+    */
+    basic_handle(const basic_handle &other) ENTT_NOEXCEPT
+        : basic_handle{other.reg, other.entt}
+    {}
+
     /**
      * @brief Constructs a handle from a given registry and entity.
      * @param ref An instance of the registry class.
      * @param value An entity identifier.
      */
-    basic_handle(registry_type &ref, entity_type value = null) ENTT_NOEXCEPT
-        : reg{&ref}, entt{value}
+    basic_handle(registry_type *registry, entity_type value = null) ENTT_NOEXCEPT
+        : reg{registry}, entt{value}
     {}
 
     /**
-     * @brief Assigns an entity to a handle.
+     * @brief Assigns an entity to a handle while leaving registry unchanged.
```

Comment:
That's good but... how do I reassign the registry?
As it stands (if I get it right, it's midnight here 🙂), if you create a handle without a registry, there is no way to turn the same in a valid handle. On the other side, if I create a handle with a registry and the null entity, I can make it a valid handle later on. Am I wrong?

Reply:
I think that reassigning the registry alone is not something you do often since most of the time it also causes entity reassignment. In such case `handle = {registry, entity}` is fine.

Still, there's nothing that prevents us from adding `operator=(basic_registry&)` or an explicit setter if you think that's desirable.